contracts/pool_stable/src/contract.rs

```diff
@@ -750,6 +750,16 @@ fn do_swap(
 ) -> i128 {
     let config = get_config(&env);
 
+    let swapped_asset_precision = token_contract::Client::new(&env, &offer_asset).decimals();
```

Comment:
We are already saving token's decimals during instantiation.
https://github.com/Phoenix-Protocol-Group/phoenix-contracts/blob/2e578a4aa7e60ddfd0a8e85ed61e1dc1c6cfd69f/contracts/pool_stable/src/storage.rs#L89-L106

use that instead of calling the client.

Reply:
And actually this should be checked and errored during the instantiation and not swap. Why would we need a pool where we can't do a single swap?

Reply:
> And actually this should be checked and errored during the instantiation and not swap. Why would we need a pool where we can't do a single swap?

absolutely, that's why I added the following check when I started working on this task:

```rs
    if precision1 > MAXIMUM_ALLOWED_PRECISION || precision2 > MAXIMUM_ALLOWED_PRECISION {
        log!(
            &env,
            "Pool Stable: Save Greatest Precision: precision above the limit"
        );
        panic_with_error!(env, ContractError::MaximumAllowedPrecisionViolated);
    }
```

we've tested it, too. 

I'll just remove the check for this during `do_swap`